src/components/OSCALMetadata.js

```diff
@@ -38,17 +59,226 @@ const OSCALMetadataPartiesCard = styled(Paper)(({ theme }) => ({
   backgroundColor: theme.palette.background.paper,
   position: "relative",
   overflow: "auto",
-  maxHeight: "12em",
+  maxHeight: "18em",
 }));
 
 // Returns a string with a locality-sensitive representation of this date
 const formatDate = (isoDate) => new Date(isoDate).toLocaleString();
 
+const TELEPHONE_ICON_MAPPING = {
+  home: <HomeIcon />,
+  mobile: <SmartphoneIcon />,
+  office: <BusinessIcon />,
+};
+
+const ADDRESS_ICON_MAPPING = {
+  home: <HomeIcon />,
+  work: <BusinessIcon />,
+};
+
+const UNKNOWN_TYPE_ICON = <HelpOutlineIcon />;
+
+function getIconOfType(mapping, contactType) {
+  if (!mapping || !contactType) {
+    return UNKNOWN_TYPE_ICON;
+  }
+  return mapping[contactType] ?? UNKNOWN_TYPE_ICON;
+}
+
+function getPhoneIcon(contactType) {
+  return getIconOfType(TELEPHONE_ICON_MAPPING, contactType);
+}
+
+function getAddressIcon(contactType) {
+  return getIconOfType(ADDRESS_ICON_MAPPING, contactType);
+}
+
+export function OSCALMetadataPartyAddress(props) {
+  const { address } = props;
+  const addrString = [
+    ...address["addr-lines"],
+    `${address.city}, ${address.state} ${address["postal-code"]}`,
+    address.country,
+  ]
+    .filter((line) => line)
+    .flatMap((item) => [item, <br key={item} />]);
+
+  return (
+    <Grid container spacing={1}>
+      <Grid item>{getAddressIcon(address.type)}</Grid>
+      <Grid item>
+        <Typography>{addrString}</Typography>
+      </Grid>
+    </Grid>
+  );
+}
+
+export function OSCALMetadataPartyEmail(props) {
+  return (
+    <Typography>
+      <Link href={`mailto:${props.email}`}>{props.email}</Link>
+    </Typography>
+  );
+}
+
+export function OSCALMetadataPartyTelephone(props) {
+  const { telephone } = props;
+  return (
+    <Typography>
+      {getPhoneIcon(telephone.type)}
+      <Link href={`tel:${telephone.number}`}>{telephone.number}</Link>
+    </Typography>
+  );
```

Comment:
Can this use a similar structure to `OSCALMetadataPartyAddress`? Maybe even a common component:

```tsx
function TextWithIcon(props) {
  const { icon, text } = props;
  return (
    <Grid container spacing={1}>
      <Grid item>{icon}</Grid>
      <Grid item>
        <Typography>{text}</Typography>
      </Grid>
    </Grid>
  );
}
```

Reply:
We used a stack!